packages/contractkit/src/wallets/ledger-wallet.test.ts

```diff
@@ -232,4 +243,107 @@ describe('LedgerWallet class', () => {
       })
     })
   })
+
+  describe('asking for addresses validations', () => {
+    describe('never', () => {
+      const knownAddress = ACCOUNT_ADDRESS1
+
+      beforeEach(() => {
+        wallet = new LedgerWallet(undefined, undefined, AddressValidation.never)
+        mockForceValidation = jest.fn((): void => {
+          // do nothing
+        })
+        mockLedger(wallet, mockForceValidation)
+      })
+
+      it("won't validate", async () => {
+        await wallet.init({})
+        expect(mockForceValidation.mock.calls.length).toBe(0)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
+        expect(mockForceValidation.mock.calls.length).toBe(0)
+      })
+    })
+
+    describe('only in the initialization', () => {
+      const knownAddress = ACCOUNT_ADDRESS1
+
+      beforeEach(() => {
+        wallet = new LedgerWallet(undefined, undefined, AddressValidation.onlyInitialization)
+        mockForceValidation = jest.fn((): void => {
+          // do nothing
+        })
+        mockLedger(wallet, mockForceValidation)
+      })
+
+      it('will validate the addresses only in the initialization', async () => {
+        await wallet.init({})
+        expect(mockForceValidation.mock.calls.length).toBe(5)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
+        expect(mockForceValidation.mock.calls.length).toBe(5)
+      })
+    })
+
+    describe('every transaction', () => {
+      const knownAddress = ACCOUNT_ADDRESS1
+
+      beforeEach(() => {
+        wallet = new LedgerWallet(undefined, undefined, AddressValidation.everyTransaction)
+        mockForceValidation = jest.fn((): void => {
+          // do nothing
+        })
+        mockLedger(wallet, mockForceValidation)
+      })
+
+      it('will validate every transaction', async () => {
+        await wallet.init({})
+        expect(mockForceValidation.mock.calls.length).toBe(0)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
+        expect(mockForceValidation.mock.calls.length).toBe(1)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
+        expect(mockForceValidation.mock.calls.length).toBe(2)
+      })
+    })
+
+    describe('once per address', () => {
+      const knownAddress = ACCOUNT_ADDRESS1
+
+      beforeEach(() => {
+        wallet = new LedgerWallet(undefined, undefined, AddressValidation.oncePerAddress)
+        mockForceValidation = jest.fn((): void => {
+          // do nothing
+        })
+        mockLedger(wallet, mockForceValidation)
+      })
+
+      it('will validate only in the first transaction of the address', async () => {
+        await wallet.init({})
+        expect(mockForceValidation.mock.calls.length).toBe(0)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
+        expect(mockForceValidation.mock.calls.length).toBe(1)
+        await wallet.signPersonalMessage(knownAddress, ACCOUNT_ADDRESS_NEVER)
```

Comment:
Can you add another check in this test that a new address increments again:
```
await wallet.signPersonalMessage(otherAddress, ACCOUNT_ADDRESS_NEVER)
expect(mockForceValidation.mock.calls.length).toBe(2)
```

Reply:
Of course!